### PR TITLE
feat: [par-4872] add pdp offer admin toggle

### PR DESCRIPTION
**Pull request by milesmcleod**
# Description

* new toggle added
* ensure it is toggled off any time a) the module is disabled or b) PP is disabled
* ensure the config and full page caches are cleaned whenever the new toggle's value is changed in order to ensure the change is immediately picked up in the UI
* new sub-group for offer display settings that should disappear if PP is disabled

## Ticket

https://helloextend.atlassian.net/browse/PAR-4260

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor (new code changes but everything functions the same)

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have deployed and validated my code on a sandbox
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: Model/Config/Backend/Enable.php [hunk 1]
```diff
@@ -56,6 +56,15 @@ public function afterSave()
...
                 0
             );
+            $this->writer->save(
+                \Extend\Integration\Service\Extend::ENABLE_PRODUCT_PROTECTION_PRODUCT_DISPLAY_PAGE_OFFER,
+                0
+            );
+
+            // need to clean these cache types in order to immediately propagate the new settings to the storefront
+            // and eliminate the need for the admin to flush the cache manually in order to see all the changes in the UI.
+            $this->cacheTypeList->cleanType('config');
```
Comment (scott-taylor-extend) [inline Model/Config/Backend/Enable.php]: This is already done in parent::afterSave(), and it's parent as well.

File: Model/Config/Backend/EnableProductProtection.php [hunk 2]
```diff
@@ -58,6 +58,15 @@ public function afterSave()
...
                 0
             );
+            $this->writer->save(
+                \Extend\Integration\Service\Extend::ENABLE_PRODUCT_PROTECTION_PRODUCT_DISPLAY_PAGE_OFFER,
+                0
+            );
+
+            // need to clean these cache types in order to immediately propagate the new settings to the storefront
+            // and eliminate the need for the admin to flush the cache manually in order to see all the changes in the UI.
+            $this->cacheTypeList->cleanType('config');
```
Comment (scott-taylor-extend) [inline Model/Config/Backend/EnableProductProtection.php]: see comment above

File: Model/Config/Backend/EnableProductProtectionProductDisplayPageOffer.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            $resourceCollection
+        );
+        $this->cacheTypeList = $cacheTypeList;
+    }
+
+    public function afterSave()
+    {
+        // need to clean these cache types in order to immediately propagate the new setting to the storefront
+        // and eliminate the need for the admin to flush the cache manually in order to see the change in the UI.
+        $this->cacheTypeList->cleanType('config');
```
Comment (scott-taylor-extend) [inline Model/Config/Backend/EnableProductProtectionProductDisplayPageOffer.php]: this shouldn't be needed either

File: Model/Config/Frontend/EnableProductProtectionProductDisplayPageOffer.php
Comment (scott-taylor-extend) [inline Model/Config/Frontend/EnableProductProtectionProductDisplayPageOffer.php:1]: What's the purpose of this file?

File: view/frontend/layout/catalog_category_view.xml [hunk 4]
```diff
@@ -8,7 +8,7 @@
 <page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
     <body>
         <referenceBlock name="category.product.addto">
-            <block class="Magento\Catalog\Block\Product\ProductList\Item\Block" after="-" ifconfig="extend_plans/product_protection/enable" name="extend.catalog.product.view.product-protection-modal-offer" template="Extend_Integration::catalog/product/view/product-protection-modal-offer.phtml">
+            <block class="Magento\Catalog\Block\Product\ProductList\Item\Block" after="-" ifconfig="extend_plans/product_protection/offer_display_settings/enable_pdp_offer" name="extend.catalog.product.view.product-protection-modal-offer" template="Extend_Integration::catalog/product/view/product-protection-modal-offer.phtml">
```
Comment (scott-taylor-extend) [inline view/frontend/layout/catalog_category_view.xml:11]: A config path should only ever be 3 parts

**Review by scott-taylor-extend — CHANGES_REQUESTED** [review]

File: view/frontend/layout/catalog_category_view.xml (hunk 4 above)
Comment (milesmcleod) [reply view/frontend/layout/catalog_category_view.xml:11]: Is there a reason for that? [It seems from the system.xml docs that groups can have nested groups](feature/PAR-4879-add-post-purchase-lead-modal-admin-toggle), and Mason indicated in the AC that there should be a nested `Offer Display Options` group within `Product Protection`

File: Model/Config/Backend/Enable.php (hunk 1 above)
Comment (milesmcleod) [reply Model/Config/Backend/Enable.php]: ok. seems like it's only the `full_page` cache that needs to be cleaned here then

File: Model/Config/Frontend/EnableProductProtectionProductDisplayPageOffer.php
Comment (milesmcleod) [reply Model/Config/Frontend/EnableProductProtectionProductDisplayPageOffer.php:1]: Had added some custom logic until I realized I only needed the `<depends/>` flag in the system.xml. Will remove it. Is the frontend model for the field in `system.xml` implied to be `\Magento\Config\Block\System\Config\Form\Field` if no `<frontend_model/>` is specified? or do i need to explicitly `<frontend_model>\Magento\Config\Block\System\Config\Form\Field</frontend_model>` for that field?

File: view/frontend/layout/catalog_category_view.xml (hunk 4 above)
Comment (milesmcleod) [reply view/frontend/layout/catalog_category_view.xml:11]: per convo with @scott-taylor-extend, this should be ok. [here are some examples of more deeply nested config paths](https://experienceleague.adobe.com/docs/commerce-operations/configuration-guide/paths/config-reference-general.html?lang=en#system-paths)

**Comment by milesmcleod** [comment]
agreed with Scott that we will remove programmatic cleaning of the full_page cache and leave it up to the merchant admin to do that manually. standby

**Review by scott-taylor-extend — APPROVED** [review]

**Review by jomarsantos — APPROVED** [review]

**Pull request merged**